superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -641,15 +641,15 @@ function nvd3Vis(element, props) {
         // If x bounds are shown, we need a right margin
         margins.right = Math.max(20, maxXAxisLabelHeight / 2) + marginPad;
       }
-      if (staggerLabels) {
-        margins.bottom = 40;
-      } else {
+      if (xLabelRotation === 45) {
```

Comment:
@kristw what was the problem with the code? I thought that since `xLabelRotation` and `staggerLabels` are mutually exclusive we could check for stagger labels first. And if no rotation is set (ie, `xLabelRotation = 0`) then lines 647-651 would have no effect since we now compute the margins based on the angle.

Reply:
Short ans: I changed and it fixes the issue with line chart. 
Longer ans: There are other code that modify `margins.right` above, which will be overridden by your `else` clause.